src/meta/src/stream/stream_manager.rs

```diff
@@ -619,8 +619,26 @@ impl GlobalStreamManager {
             .drop_source_fragments(&table_fragments_vec)
             .await;
 
+        // Drop table fragments directly.
+        mgr.fragment_manager
```

Comment:
Is this call on `drop_table_fragments_vec` an extra optimization unrelated to the this PR?

Reply:
Since we don't need to read the table fragments info from fragment manager during barrier processing, we can forward the drop here. Yes it's an extra optimization and for sql-backend we will drop them as well together with the catalogs in a single transaction. Actually it's the main purpose of introducing this refactoring.